app/uk/gov/hmrc/cgtpropertydisposals/connectors/returns/SubmitReturnsConnector.scala

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.connectors.returns
+
+import cats.data.EitherT
+import cats.syntax.order._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.libs.json._
+import uk.gov.hmrc.cgtpropertydisposals.connectors.DesConnector
+import uk.gov.hmrc.cgtpropertydisposals.connectors.returns.SubmitReturnsConnectorImpl.DesSubmitReturnRequest
+import uk.gov.hmrc.cgtpropertydisposals.http.HttpClient._
+import uk.gov.hmrc.cgtpropertydisposals.models.des.returns._
+import uk.gov.hmrc.cgtpropertydisposals.models.returns.DisposalMethod._
+import uk.gov.hmrc.cgtpropertydisposals.models.returns.{CalculatedTaxDue, CompleteReturn, SubmitReturnRequest}
+import uk.gov.hmrc.cgtpropertydisposals.models.{AmountInPence, Error}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@ImplementedBy(classOf[SubmitReturnsConnectorImpl])
+trait SubmitReturnsConnector {
+
+  def submit(returnRequest: SubmitReturnRequest)(
+    implicit hc: HeaderCarrier
+  ): EitherT[Future, Error, HttpResponse]
+
+}
+
+@Singleton
+class SubmitReturnsConnectorImpl @Inject() (http: HttpClient, val config: ServicesConfig)(implicit ec: ExecutionContext)
+    extends SubmitReturnsConnector
+    with DesConnector {
+
+  override def submit(
+    returnRequest: SubmitReturnRequest
+  )(implicit hc: HeaderCarrier): EitherT[Future, Error, HttpResponse] = {
+
+    val baseUrl: String    = config.baseUrl("returns")
```

Comment:
`baseUrl` and `returnUrl`  can go outside of this method to avoid unnecessary allocations every time this method is called